lib/Driver/UnixToolChains.cpp

```diff
@@ -55,10 +55,16 @@ toolchains::GenericUnix::addPluginArguments(const ArgList &Args,
   CompilerInvocation::computeRuntimeResourcePathFromExecutablePath(
       programPath, /*shared=*/true, pluginPath);
 
-  auto defaultPluginPath = pluginPath;
-  llvm::sys::path::append(defaultPluginPath, "host", "plugins");
+  // In-process plugin server path.
+  auto inProcPluginServerPath = pluginPath;
+  llvm::sys::path::append(inProcPluginServerPath, "host",
+                          "libSwiftInProcPluginServer.so");
```

Comment:
I am afraid this is not correct for cases of cross-compilation, like when compiling a Unix target from macOS. The `.so` assumes that the target and the compiler host are the same architecture, when they are not. This needs to be `.dylib`, `.so` or `.dll` depending on the compiler host, not the target. The same is true in `DarwinToolChains.cpp` and `WindowsToolChains.cpp`.

Reply:
This `SwiftInProcPluginServer` library is a "host" library for plugins, and is `dlopen()`ed  in the compiler. It must be the same platform/architecture as the compiler.

Reply:
But these files are related to the targets, not the compiler host.

I have a patch that I will try to upload tomorrow. I am not fully sure if the code around it is completely correct for every platform (for example the plugin path for Windows might want to be different, but it is part of the `WindowsToolChains.cpp`, which only applies when Windows is the target.

I think you can see the problem if from macOS you try to build something for an embedded target that uses some macro like `@DebugDescription`. The frontend will try to load the `.so` version of the library, instead of the correct `.dylib`.

Reply:
Oh I see what you mean. Maybe I misunderstood the meaning of this `lib/Driver/*ToolChains.cpp`.  I will look into it deeply tomorrow. Thank you  for the heads up.

Reply:
Resolved by https://github.com/swiftlang/swift/pull/74785